lib/DirectResourceAccessPass.cpp

```diff
@@ -167,19 +182,26 @@ bool clspv::DirectResourceAccessPass::RewriteAccessesForArg(Function *fn,
               dyn_cast<ConstantInt>(call->getOperand(0))->getZExtValue());
           const auto binding = uint32_t(
               dyn_cast<ConstantInt>(call->getOperand(1))->getZExtValue());
-          if (!merge_param_info({callee, set, binding, num_gep_zeroes, call}))
+          auto *resource_type = call->getOperand(6)->getType();
```

Comment:
Not on you, but maybe lib/Constants.h should get enums for the operands of the resource functions (and other special clspv builtins). The raw `6` isn't great for maintainability.

Reply:
yeah. For another CL.